pkg/intermediate/aggregate.go

```diff
@@ -331,6 +340,14 @@ func (a *AggregationProcess) addOrUpdateRecordInMap(flowKey *FlowKey, record ent
 		}
 		if !correlationRequired {
 			aggregationRecord.ReadyToSend = true
+			// if no correlation is required for inter-Node record, K8s metadata will not
+			// be filled. For Intra-Node flows and ToExternal flows, isMetaDataFilled is set
+			// to true by default.
+			if isRecordFromSrc(record) || isRecordFromDst(record) {
```

Comment:
Not your change? Do we still need `isRecordFromSrc` and `isRecordFromDst`? We already have flow type in the flow record. Can we use that to make code simple?

Reply:
Sure. We can switch to using flowType here. But function `isRecordFromSrc`, `isRecordFromDst` and `areRecordsFromSameNode` need to be kept because this information is not revealed from flowType.

Reply:
Yes we still need those methods. Moving to flowType here makes the code easily readable.